demo/components/opt-in-flyout/index.html

```diff
@@ -0,0 +1,101 @@
+			import '../../../src/components/opt-in-flyout/opt-out-reason.js';
+		</script>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta charset="UTF-8">
+		<style>
+			.flyout-wrapper {
+				height: 450px;
+				position: relative;
+				width: 100%;
+			}
+		</style>
+	</head>
+	<body unresolved>
+		<d2l-demo-page page-title="Opt-in Flyout">
+
+			<d2l-demo-snippet no-padding>
+				<template>
+					<div class="flyout-wrapper">
+						<d2l-labs-opt-in-flyout
+							open
+							title="Flyout Demo Opt-in"
+							short-description="This is a <b>short</b> description" long-description="This is a <b>long</b> description"
+							tab-position="right"
+							tutorial-link="https://www.example.com#tutorial"
+							help-docs-link="https://www.example.com#documentation">
+						</d2l-labs-opt-in-flyout>
+					</div>
+				</template>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<template>
+					<div class="flyout-wrapper">
+						<d2l-labs-opt-out-flyout
+							open
+							title="Flyout Demo Opt-out"
+							short-description="This is a <b>short</b> description"
+							long-description="This is a <b>long</b> description"
+							tab-position="right"
+							tutorial-link="https://www.example.com#tutorial"
+							help-docs-link="https://www.example.com#documentation">
+						</d2l-labs-opt-out-flyout>
+					</div>
+				</template>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<template>
+					<div class="flyout-wrapper">
+						<d2l-labs-opt-out-flyout
+							open
+							title="Flyout Demo Opt-out"
+							short-description="This is a <b>short</b> description"
+							long-description="This is a <b>long</b> description"
+							tab-position="right"
+							tutorial-link="https://www.example.com#tutorial"
+							help-docs-link="https://www.example.com#documentation">
+							<d2l-labs-opt-out-reason key="test-1" text="Test Option 1"></d2l-labs-opt-out-reason>
+							<d2l-labs-opt-out-reason key="test-2" text="Test Option 2"></d2l-labs-opt-out-reason>
+						</d2l-labs-opt-out-flyout>
+					</div>
+				</template>
+			</d2l-demo-snippet>
+
+			<d2l-demo-snippet no-padding>
+				<template>
+					<div class="flyout-wrapper">
+						<d2l-labs-opt-out-flyout
+							hide-reason
+							hide-feedback
+							open
+							title="Flyout Demo Opt-out - dialog disabled"
+							short-description="This is a <b>short</b> description"
+							long-description="This is a <b>long</b> description"
+							tab-position="right"
+							tutorial-link="https://www.example.com#tutorial"
+							help-docs-link="https://www.example.com#documentation">
+						</d2l-labs-opt-out-flyout>
+					</div>
+				</template>
+			</d2l-demo-snippet>
+
+			<script type="module">
+				[...document.querySelectorAll('d2l-labs-opt-in-flyout, d2l-labs-opt-out-flyout')].forEach(elem => {
+					['flyout-opened', 'flyout-closed', 'opt-in', 'opt-out'].forEach(eventName => {
+						// eslint-disable-next-line no-console
```

Comment:
Minor: Might be worth setting up an exception in `.eslintrc.json` for the demo folder like `core` does